tools/cas/cas.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/cachetools"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/digest"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/grpc/metadata"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var (
+	blobDigest   = flag.String("digest", "", "Digest of the blob to fetch, in HASH/SIZE format.")
+	blobType     = flag.String("type", "", "Type of proto to inspect: Action, ActionResult, Command, stdout, or stderr.")
+	instanceName = flag.String("remote_instance_name", "", "Remote instance name")
+	target       = flag.String("target", "", "Cache grpc target, such as grpcs://remote.buildbuddy.io")
+	apiKey       = flag.String("api_key", "", "API key to attach to the outgoing context")
+
+	invocationID = flag.String("invocation_id", "", "Invocation ID. This is required when fetching the result of a failed action. Otherwise, it's optional.")
+)
+
+func main() {
```

Comment:
would be nice to have a comment showing example usage

Reply:
Done